### ISSUE TITLE
Weird Redstone Behaviour

### ISSUE THREAD

**Issue by Xenoxis** [cuberite/cuberite#4605]
Client version: 1.12.2
Server OS: Ubuntu server 16.04 i386
Cuberite Commit id: 318d31dc394da9faea11544ad6e485f1b5eabbdb

### Expected behavior
The repeater should update the torch and set it off

### Actual behavior
The torch seems to ignore the redstone signal from the repeater (no other redstone signals comes to the torch)
After a server restart and a broke/place the repeater, it works 
![2020-04-03_11 33 18](https://user-images.githubusercontent.com/19935648/78346744-c135cd80-759f-11ea-9d85-c39de0bcdbcc.png)
![2020-04-03_11 33 26](https://user-images.githubusercontent.com/19935648/78346769-c98e0880-759f-11ea-9eca-c6bcb7a781a8.png)

### Steps to reproduce the behavior
Use redstone (in particular this setup with repeater and redstone torch) and play until it doesn't work anymore.

**Comment by Xenoxis** [comment]
![2020-04-03_19 53 52](https://user-images.githubusercontent.com/19935648/78390459-e993eb00-75e4-11ea-92a8-f6c062b632e0.png)

**Comment by tigerw** [comment]
Can you get the torch to stop working first time or do you have to toggle it a couple of times before it stops responding?

**Comment by Xenoxis** [comment]
Yes this happen after activating several times the system, and even if i broke the torch, the lamp is not responding and doesn't switch off

**Comment by tigerw** [comment]
Thanks. Do you have a world exhibiting this that I could test with?

**Comment by AirOne01** [comment]
> Thanks. Do you have a world exhibiting this that I could test with?

You can see this behavior in a lot of different scenarios by just fiddling around with redstone.
I think power conduction is broken in general. Injecting power to a block with a repeater or comparator will generally not work.

**Comment by tigerw** [comment]
Please take a pic, I've been entirely unable to reproduce

**Comment by AirOne01** [comment]
> Please take a pic, I've been entirely unable to reproduce

So I just did a big crash test and these are all the missed/broken redstone behavior I could witness. To the left, vanilla 1.12.2, and to the right, Cuberite (d49ce751ba6f8f2eb171633cea30a1119935f8c7). As I am new to contributions, I don't know what should be a separated issue. Don't hesitate to correct me if anything is wrong.
1. Here in vanilla you can use a bug with slabs that allow this kind of redstone elevator. The bug is not present in Cuberite so that was to be expected. ![image](https://user-images.githubusercontent.com/21955960/141846267-5bac2eb5-71ff-48db-b797-40cfa2fa9e0a.png)
2. Here is a simple slime block contraption that use BUD i believe. It activated and then deactivated the piston every time a block updated next to the piston. It was before observers were added of course, but it still works. In Cuberite, the piston just activates and deactivates rapidly enough for the animation to be skipped. ![image](https://user-images.githubusercontent.com/21955960/141847206-7cd2131f-2671-4829-90fd-08c7d9367f81.png)
3. I found another bug with the repeater which didn't activated once i sticked a redstone block behind it. I tried again with a torch and it worked, then again with a redstone block and it didn't work. I then destroyed the blocks to try some things but couldn't reproduce the bug. I tried different directions, coordinates, etc, but couldn't get it to glitch again. Not sure if this helps.
4. Dropper don't drop their content when broken. I tried breaking them in creative, survival and with explosions.
5. It seems you can place some illegal blocks on pistons, but they will not behave the same as usual. Levers can't be activated and redstone can't be powered. Your item won't be consumed (which means placed block is client-side). I noticed some interresting behavior with buttons.
![image](https://user-images.githubusercontent.com/21955960/141850582-f35956ea-a806-462c-aea2-297d4ed08333.png)
You can press it (it will stay pressed), and then right click it with a block to replace it with this block. Shift-right clicking will replace the block and place another one, thus duplicating it *client-side*. This can be used to create ghost blocks (and fly).
![image](https://user-images.githubusercontent.com/21955960/141856808-f8d2097e-5f3e-4c89-9b77-11156e2e936a.png)
Here the result on two account. Both were connected along the process but only the evildoer's client can see the block.
Appart from that, you can also place ladders alongside pistons by right-clicking. They will be placed server-side and will show up for any player.
6. If you spam a lever, you can make visual errors (I found another one similar to this one). This is patched by disconnecting and joining again or by reloading the chunk.
![image](https://user-images.githubusercontent.com/21955960/141860612-56ba6900-ea25-4715-b395-9c373d78b408.png)
7. In this setup where two pistons fight for expansion, only one is supposed to take priority, but the other one activates after being unpowered. this may have something to do with the piston beeing updated after the other one. In vanilla, only on etakes the priority. Also it seems to only work on the X axis. ![2021-11-15 22-49-36](https://user-images.githubusercontent.com/21955960/141861825-6bd27fe4-96e6-4337-bcd8-0c83059096e0.gif)
8. The next setup (two sticky pistons facing each other)
![ezgif-4-58b6e804fa55](https://user-images.githubusercontent.com/21955960/141863430-34af5adc-ec2c-4af0-b040-c1129b3d0464.gif)
I don't know how this is possible. It's not client-sided only. I assumed it had something to do with BUDs, but you can't trigger the piston update by updating any block next to it. Only by removing the redstone block. I couldn't reproduce it with redstone torches and redstone injection.
9. Turning the lever on in this contraption causes a BUD in vanilla, and the only way to update the piston is to update any of the blocks directly next to it. In Cuberite, you can update any of the orange-marked blocks in this image to trigger the update, as well as destroy and replace the green block to update the redstone.
![image](https://user-images.githubusercontent.com/21955960/141864226-567f545e-95c7-4321-a92e-a66ca3a7a395.png)

**Comment by tigerw** [comment]
Thanks, this is indeed super helpful! Did you see any misbehaviour when powering blocks with redstone wire or repeaters?

**Comment by AirOne01** [comment]
> Thanks, this is indeed super helpful! Did you see any misbehaviour when powering blocks with redstone wire or repeaters?

Not directly, no. The only time i had a problem with direct powering was no. 3 of my comment. I also couldn’t get to find any direct duplication glitch. Pretty much all of my researches were focused on these.
To summary I would say most misbehaviours are about pistons and BUDs.
